File: 2038-如果相邻两个颜色均相同则删除当前颜色.cpp
#include <iostream>
using namespace std;
#include <string>

class Solution
{
public:
    bool winnerOfGame(string colors)
    {
        int count_a = 0, count_b = 0, num_a = 0, num_b = 0;
        for (char ch : colors)
        {
            if (ch == 'A')
            {
                count_b = 0;
                count_a++;
                if (count_a >= 3)
                    num_a++;
            }
            else
            {
                count_a = 0;
                count_b++;
                if (count_b >= 3)
                    num_b++;
            }
        }
        return num_a > num_b;
    }
};

File: 2038-如果相邻两个颜色均相同则删除当前颜色.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        count_a = count_b = num_a = num_b = 0
        for ch in colors:
            if ch == "A":
                count_b = 0
                count_a += 1
                if count_a >= 3:
                    num_a += 1
            else:
                count_a = 0
                count_b += 1
                if count_b >= 3:
                    num_b += 1
        return num_a > num_b
